dhcp/pxe.go

```diff
@@ -88,13 +86,38 @@ func IsPXE(req *dhcp4.Packet) bool {
 	return ok && strings.HasPrefix(class, "PXEClient")
 }
 
-func SetupPXE(rep, req *dhcp4.Packet) bool {
+func SetupPXE(ctx context.Context, rep, req *dhcp4.Packet) bool {
 	if !copyGUID(rep, req) {
 		if class, ok := req.GetString(dhcp4.OptionClassID); !ok || !strings.HasPrefix(class, "PXEClient") {
 			return false // not a PXE client
 		}
 		dhcplog.With("mac", req.GetCHAddr(), "xid", req.GetXID()).Info("no client GUID provided")
 	}
+
+	/*
+		Intel's Preboot Execution Environment (PXE) Specification (1999):
+
+		Options 1-63 are reserved for PXE, we use option 6 with value 0x8:
+
+			These options control the type of boot server discovery mechanisms used
+			by clients. Clients must use discovery methods in this order:
+			[options 1 & 2 omitted bc they're irrelevant to us]
+
+			3. Unicast. If a Boot Server list is available, (PXE_BOOT_SERVERS,
+			   Option #43 tag #8). If PXE_DISCOVERY_CONTROL bit 2 is set, the client
+			   may still use multicast and broadcast discovery (if it is permitted
+			   by bits 0 and 1); but the client may only accept replies from servers
+			   that are identified in the PXE_BOOT_SERVERS option.
+
+		Options 64-127 are "boot server specific" so that's why we put traceparent
+		propagation in opt43/slot69.
+	*/
+
+	pxeVendorOptions := dhcp4.OptionMap{
```

Comment:
I really like that you moved this to be defined where it's used instead of it being a package level variable. Not to mention the great code comment. Thank you!

Reply:
Thank the Intel PXE specification :) I found that bit on the way to understanding option43 and it seemed worthwhile to inline.